src/main/java/org/wildfly/security/password/impl/BSDUnixDESCryptPasswordImpl.java

```diff
@@ -293,20 +293,20 @@ private static void setupTables() {
 
         // Compute the initial and final permutations and also initialize the inverted key permutation
         for (int i = 0; i < 64; i++) {
-            finalPerm[i] = (byte) (IP[i] - 1);
-            initPerm[finalPerm[i]] = (byte) i;
-            invKeyPerm[i] = (byte) 255;
+            finalPerm[i] = (IP[i] & 0xff) - 1;
```

Comment:
This should be preceeded by:

```
assert IP[i] != 0;
```
Or else the logic should be corrected to:
```
finalPerm[i] = (IP[i] - 1) & 0xff;
```
Otherwise finalPerm[i] might end up being 0xffffffff.